spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinition.java

```diff
@@ -38,24 +41,35 @@
 	private final String dslText;
 
 	public TaskDefinition(String name, String dsl) {
+		this(name, dsl, false);
+	}
+
+	TaskDefinition(String registeredAppName, String label, Map<String, String> properties) {
+		super(registeredAppName, label, properties);
+		this.dslText = "";
+		properties.put("spring.cloud.task.name", registeredAppName);
+	}
+
+	public TaskDefinition(String name, String dsl, boolean isComposedTask) {
 		this.dslText = dsl;
-		AppNode taskNode = new TaskParser(name, dsl).parse();
-		setRegisteredAppName(taskNode.getName());
 		Map<String, String> properties = new HashMap<>();
-		if (taskNode.hasArguments()) {
-			for (ArgumentNode argumentNode : taskNode.getArguments()) {
-				properties.put(argumentNode.getName(), argumentNode.getValue());
+		if(isComposedTask) {
```

Comment:
I think the primary issue is that a task definition is comprised of an app with supporting properties.  While a composed task is comprised of task definitions.